include/alpaka/mem/view/AccessorAdaptor.hpp

```diff
@@ -0,0 +1,169 @@
+            using const_reference = value_type const&;
+
+        public:
+            AccessorAdaptor()
+            {
+                static_assert(experimental::traits::internal::IsView<TView>::value, "");
+            }
+
+            ALPAKA_FN_HOST auto data() -> pointer
+            {
+                return getPtrNative(*static_cast<TView*>(this));
+            }
+
+            ALPAKA_FN_HOST auto data() const -> const_pointer
+            {
+                return getPtrNative(*static_cast<TView const*>(this));
+            }
+
+            ALPAKA_FN_HOST auto operator*() -> reference
+            {
+                static_assert(
+                    alpaka::Dim<TView>::value == 0,
+                    "operator* is only valid for Buffers and Views of dimension 0");
+                return *data();
+            }
+
+            ALPAKA_FN_HOST auto operator*() const -> const_reference
+            {
+                static_assert(
+                    alpaka::Dim<TView>::value == 0,
+                    "operator* is only valid for Buffers and Views of dimension 0");
+                return *data();
+            }
+
+            ALPAKA_FN_HOST auto operator->() -> pointer
+            {
+                static_assert(
+                    alpaka::Dim<TView>::value == 0,
+                    "operator-> is only valid for Buffers and Views of dimension 0");
+                return data();
+            }
+
+            ALPAKA_FN_HOST auto operator->() const -> const_pointer
+            {
+                static_assert(
+                    alpaka::Dim<TView>::value == 0,
+                    "operator-> is only valid for Buffers and Views of dimension 0");
+                return data();
+            }
+
+            ALPAKA_FN_HOST auto operator[](std::size_t i) -> reference
+            {
+                static_assert(
+                    alpaka::Dim<TView>::value == 1,
+                    "operator[i] is only valid for Buffers and Views of dimension 1");
+                return data()[i];
+            }
+
+            ALPAKA_FN_HOST auto operator[](std::size_t i) const -> const_reference
+            {
+                static_assert(
+                    alpaka::Dim<TView>::value == 1,
+                    "operator[i] is only valid for Buffers and Views of dimension 1");
+                return data()[i];
+            }
+
+        private:
+            template<std::size_t TDim, typename TIdx>
+            ALPAKA_FN_HOST auto ptr_at(Vec<DimInt<TDim>, TIdx> index) const -> const_pointer
+            {
+                using Idx = alpaka::Idx<TView>;
+                static_assert(
+                    alpaka::Dim<TView>::value == TDim,
+                    "the index type must have the same dimensionality as the Buffer or View");
+                static_assert(
+                    std::is_convertible_v<TIdx, Idx>,
+                    "the index type must be convertible to the index of the Buffer or View");
```

Comment:
You could just use the `TView`'s dimensionality and avoid the check:

```suggestion
            template<typename TIdx>
            ALPAKA_FN_HOST auto ptr_at(Vec<alpaka::Dim<TView>, TIdx> index) const -> const_pointer
            {
                using Idx = alpaka::Idx<TView>;
                static_assert(
                    std::is_convertible_v<TIdx, Idx>,
                    "the index type must be convertible to the index of the Buffer or View");
```

Reply:
Same problem as above: `alpaka::Dim<TView>` does not work during the class definition, so it cannot be used as part of the method signature.

Reply:
Ok let's leave it as is for now.